Лабораторная работа 1

BagovaMYu opened the pull request.

BagovaMYu committed:
Лабораторная работа 1

andreiivanov-wa commented on main.py:
```diff
@@ -0,0 +1,208 @@
+    >>> сhristmas_tree = ChristmasTree(30, {'balloon': 10, 'star': 1})
+    """
+    def __init__(self, slots_for_decor: int, decor: dict):
+        if not isinstance(slots_for_decor, int):
```
описания нет

andreiivanov-wa commented on main.py:
```diff
@@ -0,0 +1,208 @@
+        Примеры:
+        >>> сhristmas_tree = ChristmasTree(30, {'balloon': 10, 'star': 1})
+        >>> сhristmas_tree.free_slots_check()
+        """
```
pass или ...

andreiivanov-wa commented on main.py:
```diff
@@ -0,0 +1,208 @@
+    >>> bulb = Bulb(60, True)
+    """
+    def __init__(self, power: int, light_is_on: bool):
+        self.power_availability_check(power)
```
описание

andreiivanov-wa commented on main.py:
```diff
@@ -0,0 +1,208 @@
+
+        :raise ValueError: Если указано нестандартное значение мощности, то вызываем ошибку
+        """
+        standart_power = [20, 40, 60, 75, 100]
```
это свойство класса, а не переменная функции

BagovaMYu replied:
@andreiivanov-wa свойства класса будем проходить в теме 3, пока разобраться не получилось, оставила так

andreiivanov-wa replied:
просто внутренней переменной класса сделать. не надо ее создавать внутри функции, это же "константа" для класс, в конструкторе __init__ определелить
self.standart_power = [20, 40, 60, 75, 100]
и использовать в коде

BagovaMYu replied:
А, поняла. Исправила

andreiivanov-wa commented on main.py:
```diff
@@ -0,0 +1,208 @@
+        if not isinstance(power, int):
+            raise TypeError("Мощность лампочки должна быть типа int")
+        if power not in standart_power:
+            raise ValueError("Допустимое значение мощности 20, 40, 60, 75 или 100")
```
а если в будущем программист внесет изменение в список standart_power и забудет изменить в строке 156 текстовое описание? нужно list to str сделать

BagovaMYu committed:
Update main.py

Свойства класса будем проходить в теме 3, пока не получилось разобраться. Остальное, вроде, исправила

andreiivanov-wa commented on main.py:
```diff
@@ -0,0 +1,205 @@
+
+
+class ChristmasTree:
+    def __init__(self, slots_for_decor: int, decor: dict):
```
для словаря тоже можно типы ключа и значения задать

BagovaMYu committed:
Update main.py
